### PR TITLE
Fix bug with goals table print goals

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Found while testing out the prior implementation of select checkboxes > print to PDF. Didn't want to pollute the parent PR by including this fix there

## How to test
Go to RTR > Goals and objectives
Select a handful of goals. 
Click preview and print selected
Confirm your selection is all that is displayed

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete
- [ ] QA review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] PR created as **Draft**
- [ ] Staging smoke test completed
- [ ] PR transitioned to **Open**
- [ ] Reviewer added _(after transitioning to Open to ensure Slack notifications trigger)_
  - _Sequence: Draft PR → Smoke test → Open PR → Add reviewer_
  - _Confirm that Slack notification was sent after reviewer was added_

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/services/standardGoals.ts [hunk 1]
```diff
@@ -843,11 +843,13 @@ export async function standardGoalsForRecipient(
   onlyApprovedObjectives = false,
 ) {
   const { goal: scopes } = await filtersToScopes(filters, {});
+  const validGoalIds = [goalIds].flat().filter((id) => !Number.isNaN(Number(id)));
 
   const goals = await Goal.findAll({
     attributes: ['id'],
     where: {
       [Op.and]: [
+        ...(validGoalIds.length ? [{ id: validGoalIds }] : []),
```
Comment (kryswisnaskas) [inline src/services/standardGoals.ts:870]: Claude said: **Finding 1 — Invalid `goalIds` silently returns all goals instead of none**

When every submitted ID fails the `Number.isNaN` check (e.g. `?goalIds=abc`), `validGoalIds` ends up empty and this condition skips the filter entirely — returning every goal for the recipient. That is indistinguishable from the "no filter requested" case (`goalIds` defaulting to `[]`), so a corrupted or non-numeric selection in the print view silently shows the full list instead of surfacing an error.

Additionally, `Number("") === 0` and `Number(null) === 0` both pass `!Number.isNaN(...)`, so an empty-string query param (`?goalIds=`) is included as ID `0`, which matches no real rows — producing an empty print view with no indication of why.

Suggested fix:
```ts
const rawGoalIds = [goalIds].flat();
const validGoalIds = rawGoalIds.filter(id => Number.isInteger(Number(id)) && Number(id) > 0);

// If the caller explicitly passed IDs but none were valid, return empty rather than all goals
if (rawGoalIds.length && !validGoalIds.length) {
  return { count: 0, goalRows: [], statuses: { total: 0, ... }, allGoalIds: [] };
}
```

File: src/services/standardGoals.ts [hunk 2]
```diff
@@ -843,11 +843,13 @@ export async function standardGoalsForRecipient(
   onlyApprovedObjectives = false,
 ) {
   const { goal: scopes } = await filtersToScopes(filters, {});
+  const validGoalIds = [goalIds].flat().filter((id) => !Number.isNaN(Number(id)));
```
Comment (kryswisnaskas) [inline src/services/standardGoals.ts]: Claude said: **Finding 3 — No test coverage for the new `goalIds` filtering behavior**

The existing integration test suite in `src/services/standardGoal.test.js` already creates goals with known IDs inside a `beforeAll` block, so tests for this new branch would be low-effort to add. The scenarios that need coverage:

- Passing a single valid `goalId` returns only that goal (not others for the same recipient)
- Passing a subset of valid IDs returns only that subset
- Passing all-invalid IDs (e.g. `'abc'`) does not silently return all goals
- Passing an empty array (the default) continues to return all goals unchanged

Without these, the silent-fallback bug in Finding 1 would not have been caught by tests even if it had been written as a regression.

**Comment by kryswisnaskas** [comment]
Trying out an additional review pass using AI (Claude/Codex) to help catch edge cases and things that are easy to miss.

Goal here is to speed up reviews and catch things earlier. Open to feedback if this becomes noisy or not useful.

File: src/services/standardGoals.ts (hunk 1 above)
Comment (thewatermethod) [reply src/services/standardGoals.ts:870]: In most cases there are no valid IDs sent, since the print view and the RTTAPA both use this service. This is implemented as intended

Comment (thewatermethod) [reply src/services/standardGoals.ts:870]: Well, I went ahead and added the suggestion: e89b19e

File: src/services/standardGoals.ts (hunk 2 above)
Comment (thewatermethod) [reply src/services/standardGoals.ts]: 6592b4d

**Review by kryswisnaskas — APPROVED** [review]
👍

**Review by tommaroh — APPROVED** [review]
